realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -100,7 +100,7 @@
  * As you can see, after deletion, the size and elements order of snapshot stay the same as before. But the element at
  * the position becomes invalid.
  */
-public interface OrderedRealmCollection<E extends RealmModel> extends List<E>, RealmCollection<E> {
+public interface OrderedRealmCollection<E> extends List<E>, RealmCollection<E> {
```

Comment:
As you know, I just hate doing this.  Moving these checks out of the type system and into the runtime (I *assume* we'll check at runtime!) just seems like a perversion of the typing system and a misunderstanding of Java.  I wish we could have discussed this approach as a design and considered other approaches before the coding started.  At this point, it seems too late to object.  I cannot shake the feeling, though, that we are headed for a type-less library.

Reply:
You will check it in the annotation processor, which is compilation time. Or at least I think that is the plan. 

You can't really avoid this if you ever want to introduce type adapters either, so I think this is a good decision overall.